### PR TITLE
[ASM][IAST] New scrubbing for location data

### PR DESCRIPTION
**Pull request by e-n-0**
## Summary of changes

Update all snapshots:
- With a correct location data
- Without git tags that can't be generated without symbols

## Reason for change

In integration tests for IAST, the vulnerability location is completely removed because it can vary between FW versions. We want a way of keeping locations getting rid of this variance, by scrubbing the data in a smart way.

## Implementation details

- Removing all symbols for consistencies between all archs and frameworks.
    - Side effect: Removing all git tags that are inside pdbs

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5047)](https://github.com/DataDog/dd-trace-dotnet/tree/51560308c9c56ce2b2eb40a2aa4bc6d6874b5790)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/223293907dfbfd9f960ac0b84b9127ae7c9f2560)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (71ms)  : 62, 80
     .   : milestone, 71,
    master - mean (71ms)  : 60, 81
     .   : milestone, 71,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (1,027ms)  : 1003, 1052
     .   : milestone, 1027,
    master - mean (1,024ms)  : 1002, 1047
     .   : milestone, 1024,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (105ms)  : 101, 110
     .   : milestone, 105,
    master - mean (105ms)  : 99, 111
     .   : milestone, 105,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (729ms)  : 712, 747
     .   : milestone, 729,
    master - mean (726ms)  : 704, 748
     .   : milestone, 726,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (89ms)  : 85, 94
     .   : milestone, 89,
    master - mean (88ms)  : 83, 92
     .   : milestone, 88,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (689ms)  : 671, 708
     .   : milestone, 689,
    master - mean (689ms)  : 663, 715
     .   : milestone, 689,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (187ms)  : 184, 190
     .   : milestone, 187,
    master - mean (188ms)  : 185, 190
     .   : milestone, 188,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (1,138ms)  : 1118, 1159
     .   : milestone, 1138,
    master - mean (1,142ms)  : 1122, 1163
     .   : milestone, 1142,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (271ms)  : 267, 275
     .   : milestone, 271,
    master - mean (271ms)  : 267, 275
     .   : milestone, 271,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (1,098ms)  : 1067, 1129
     .   : milestone, 1098,
    master - mean (1,102ms)  : 1080, 1124
     .   : milestone, 1102,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5047) - mean (261ms)  : 257, 264
     .   : milestone, 261,
    master - mean (260ms)  : 258, 263
     .   : milestone, 260,

    section CallTarget+Inlining+NGEN
    This PR (5047) - mean (1,059ms)  : 1036, 1082
     .   : milestone, 1059,
    master - mean (1,063ms)  : 1045, 1080
     .   : milestone, 1063,

```

**Comment by andrewlock** [comment]
## Throughput/Crank Report:zap:

Throughput results for AspNetCoreSimpleController comparing the following branches/commits:
- [This PR (5047)](https://github.com/DataDog/dd-trace-dotnet/tree/17fc32a11490bb06b30f4a9967210b541c65a153)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/a9786932a0ce03155f13b49c9a67446694b75d86)
- [benchmarks/2.9.0](https://github.com/DataDog/dd-trace-dotnet/tree/89ae3f4ee7d85b824badda5873166707c9e560b4)

Cases where throughput results for the PR are worse than latest master (5% drop or greater), results are shown in **red**.

Note that these results are based on a _single_ point-in-time result for each branch. For full results, see [one](https://ddstaging.datadoghq.com/dashboard/fnz-afi-c2i/apm-net-crank) of the [many](https://ddstaging.datadoghq.com/dashboard/c5g-i7d-kcy/ciapp-apm-net-throughput-tests), [many](https://ddstaging.datadoghq.com/dashboard/uxh-m8j-qhi/ciapp-apm-net-throughput-tests-kevin) dashboards!

```mermaid
gantt
    title Throughput Linux x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5047) (11.461M)   : 0, 11461173
    master (11.893M)   : 0, 11892622
    benchmarks/2.9.0 (11.571M)   : 0, 11571320

    section Automatic
    This PR (5047) (7.961M)   : 0, 7961368
    master (8.099M)   : 0, 8098638
    benchmarks/2.9.0 (8.323M)   : 0, 8322791

    section Trace stats
    This PR (5047) (8.319M)   : 0, 8319226
    master (5.895M)   : 0, 5895492

    section Manual
    This PR (5047) (9.982M)   : 0, 9981823
    master (10.361M)   : 0, 10361124

    section Manual + Automatic
    This PR (5047) (7.520M)   : 0, 7520193
    master (7.670M)   : 0, 7670088

    section Version Conflict
    This PR (5047) (6.766M)   : 0, 6766310
    master (6.859M)   : 0, 6859229

```
```mermaid
gantt
    title Throughput Linux arm64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5047) (9.762M)   : 0, 9761691
    master (9.617M)   : 0, 9616856
    benchmarks/2.9.0 (9.660M)   : 0, 9659654

    section Automatic
    This PR (5047) (6.745M)   : 0, 6745342
    master (6.615M)   : 0, 6615394

    section Trace stats
    This PR (5047) (6.648M)   : 0, 6647613
    master (6.934M)   : 0, 6934299

    section Manual
    This PR (5047) (8.244M)   : 0, 8243750
    master (8.540M)   : 0, 8539968

    section Manual + Automatic
    This PR (5047) (6.024M)   : 0, 6023565
    master (6.148M)   : 0, 6147793

    section Version Conflict
    This PR (5047) (5.681M)   : 0, 5681124
    master (5.693M)   : 0, 5693193

```
```mermaid
gantt
    title Throughput Windows x64 (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5047) (10.266M)   : 0, 10265791
    benchmarks/2.9.0 (9.963M)   : 0, 9963205

    section Automatic
    This PR (5047) (7.381M)   : 0, 7380543
    benchmarks/2.9.0 (7.483M)   : 0, 7483414

    section Trace stats
    This PR (5047) (7.694M)   : 0, 7694116

    section Manual
    This PR (5047) (9.343M)   : 0, 9343358

    section Manual + Automatic
    This PR (5047) (7.019M)   : 0, 7018981

    section Version Conflict
    This PR (5047) (6.227M)   : 0, 6226521

```
```mermaid
gantt
    title Throughput Linux x64 (ASM) (Total requests) 
    dateFormat  X
    axisFormat %s
    section Baseline
    This PR (5047) (7.429M)   : 0, 7428814
    master (7.496M)   : 0, 7496114
    benchmarks/2.9.0 (7.887M)   : 0, 7887276

    section No attack
    This PR (5047) (1.747M)   : 0, 1746664
    master (1.764M)   : 0, 1763502
    benchmarks/2.9.0 (3.249M)   : 0, 3249034

    section Attack
    This PR (5047) (1.427M)   : 0, 1427231
    master (1.399M)   : 0, 1399328
    benchmarks/2.9.0 (2.565M)   : 0, 2565044

    section Blocking
    This PR (5047) (3.120M)   : 0, 3120423
    master (3.098M)   : 0, 3097902

    section IAST default
    This PR (5047) (6.503M)   : 0, 6503112
    master (6.459M)   : 0, 6458516

    section IAST full
    This PR (5047) (5.710M)   : 0, 5709596
    master (5.769M)   : 0, 5769139

    section Base vuln
    This PR (5047) (1.007M)   : 0, 1006708
    master (0.971M)   : 0, 970670

    section IAST vuln
    This PR (5047) (0.856M)   : 0, 856431
    master (0.901M)   : 0, 900962

```

File: tracer/test/test-applications/security/Samples.Security.AspNetCore2/Samples.Security.AspNetCore2.csproj [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
   <PropertyGroup>
     <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
+    <DebugType>None</DebugType>
```
Comment (NachoEchevarria) [inline tracer/test/test-applications/security/Samples.Security.AspNetCore2/Samples.Security.AspNetCore2.csproj:5]: Why this change is needed? I guess that this change could potentially avoid line number and files being retrieved, right?

File: tracer/test/snapshots/Iast.HeaderInjection.AspNetCore5.Vuln.Cookie.SensitiveValue.verified.txt [hunk 2]
```diff
@@ -73,9 +71,7 @@
       component: aspnet_core,
       env: integration_tests,
       language: dotnet,
-      span.kind: server,
```
Comment (NachoEchevarria) [inline tracer/test/snapshots/Iast.HeaderInjection.AspNetCore5.Vuln.Cookie.SensitiveValue.verified.txt:76]: Changing the debug properties of the project makes this information disappear from the snapshots? I mean, it's fine if it's not shown, I'm just curious

**Review by NachoEchevarria — APPROVED** [review]
LGTM. I just have some questions.

File: tracer/test/test-applications/security/Samples.Security.AspNetCore2/Samples.Security.AspNetCore2.csproj (hunk 1 above)
Comment (e-n-0) [reply tracer/test/test-applications/security/Samples.Security.AspNetCore2/Samples.Security.AspNetCore2.csproj:5]: Yes the lines are not retrieved anymore.
The choice of disabling all symbols on security samples was to get a consistent data between within all tests.
The main issue was that, depending of the architecture and the OS, the details of frames in the stacktraces doesn't includes all data (failing to get the filename, then the line number, making a lot of inconsistencies).



By disabling all symbols, all archs and platform produce the same data, verified by the snapshots.

File: tracer/test/snapshots/Iast.HeaderInjection.AspNetCore5.Vuln.Cookie.SensitiveValue.verified.txt (hunk 2 above)
Comment (e-n-0) [reply tracer/test/snapshots/Iast.HeaderInjection.AspNetCore5.Vuln.Cookie.SensitiveValue.verified.txt:76]: The `span.kind: server` isn't removed here, only the comma at the end is removed because it's now the last tag, because the last 2 tags are removed. The diff is broken.

If you were talking about the `_dd.git.commit.sha` and `_dd.git.repository_url`, these both tags are generated with the needs of symbols. But in our case we stop the generation to get a consistent data between all frameworks, archs and plateforms (cf my other comment). That's why the doesn't get produced anymore 😄

**Review by lucaspimentel — APPROVED** [review]
